napari/utils/io.py

```diff
@@ -290,13 +291,59 @@ def write_csv(
             writer.writerow(row)
 
 
+def csv_to_layer_data(filename: str) -> List[FullLayerData]:
+    """Return layer data from a csv.
+
+    Parameters
+    ----------
+    filename : str
+        Filename for reading csv.
+
+    Returns
+    -------
+    list of napari.types.FullLayerData or None
+        Layer full layer data or None.
+    """
+    # If extension is not csv then cannot read file
+    if not filename.endswith('.csv'):
+        return None
+
+    gen = iter_csv(filename)
+    column_names = next(gen)
+    layer_type = guess_layer_type_from_column_names(column_names)
+    if layer_type:
+        print('asdf')
+        print(csv_reader_functions[layer_type])
+        return [csv_reader_functions[layer_type](filename)]
+    else:
+        return None
```

Comment:
```suggestion
        return []
```
(`ReaderFunctions` don't return `Optional[List[LayerData]]`)